Userland/Libraries/LibWeb/Painting/PaintableBox.cpp

```diff
@@ -741,6 +741,15 @@ Optional<HitTestResult> PaintableWithLines::hit_test(CSSPixelPoint position, Hit
     if (!layout_box().children_are_inline() || m_fragments.is_empty())
         return PaintableBox::hit_test(position, type);
 
+    for (auto* child = first_child(); child; child = child->next_sibling()) {
+        auto result = child->hit_test(position, type);
+        if (!result.has_value())
+            continue;
+        if (!result->paintable->visible_for_hit_testing())
+            continue;
```

Comment:
It feels weird that we check `visible_for_hit_testing()` *after* `hit_test()` returns it. Why would `hit_test()` return an unhittable node in the first place?
Not saying this is something we need to fix here, but maybe something we can improve in the future :sweat_smile: